File: DSA/HR/grading_students.cpp
#include <vector>
using namespace std;

vector<int> gradingStudents(vector<int> grades)
{
    std::vector<int> ans{};

    for (int grade : grades)
    {
        if (grade < 38)
            ans.push_back(grade);
        else if (grade % 5 >= 3)
            ans.push_back(grade + (5 - grade % 5));
        else
            ans.push_back(grade);
    }

    return ans;
}

File: DSA/HR/grading_students.java
import java.util.List;

class Result {
    public static List<Integer> gradingStudents(List<Integer> grades) {
        for (int i = 0; i < grades.size(); i++) {
            Integer grade = grades.get(i);
            if (grade >= 38 && ((grade % 5) >= 3)) {
                grades.set(i, grade + (5 - (grade % 5)));
            }
        }

        return grades;
    }
}